src/main/java/edu/illinois/cs/dt/tools/detection/filters/ConfirmationFilter.java

```diff
@@ -58,8 +59,9 @@ public boolean keep(final DependentTest dependentTest, final int absoluteRound)
                 return confirmation(true, "confirmation-sampling", dependentTest, absoluteRound);
             }
 
-            // This test is known to be flaky, so it should never make it past the filter
-            return false;
+            // This test is known to be flaky, so set its type to NOD but still keep the test
+            dependentTest.setType(DependentTestType.NOD);
+            return true;
```

Comment:
Can you explain the logic of why prior to the changes, iDFlakies was "removing" flaky tests?

The removal doesn't mean that iDFlakies doesn't run the test, right? But perhaps iDFlakies just doesn't consider the "removed" tests as OD?